### ISSUE TITLE
Vega 56 CN-GPU low hashate (auto config)

### ISSUE THREAD

**Issue by Ryo-RU** [fireice-uk/xmr-stak#2457]
**TLDR:** auto config for CN-GPU is single threaded and produces too low hr. The workaround is to use either 2.8.0 or dual thread with later releases or use newest release and set worksize to 16.

(Tested on 2 different GPUs)
Vega 56, unflashed, stock OC settings
Algo: Cn-GPU (Ryo)

**Stak 2.10.5** (release binaries) + Windows 10 (drivers doesn't matter - tried a lot of them)
Auto generated config for cryptonight-gpu 
```
  // gpu: gfx900  compute units: 56
  // memory:4048|4048|8048 MiB (used per thread|max per alloc|total free)
  { "index" : 1,
    "intensity" : 1792, "worksize" : 8,
    "affine_to_cpu" : false, "strided_index" : 0, "mem_chunk" : 2,
    "unroll" : 1, "comp_mode" : true, "interleave" : 40
  },
```
**Log:**
```
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m |
|  0 | 1151.8 |   (na) |   (na) |
Totals (AMD):  1151.8    0.0    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1151.8    0.0    0.0 H/s
Highest:  1163.4 H/s
-----------------------------------------------------------------
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m |
|  0 | 1158.7 | 1154.0 |   (na) |
Totals (AMD):  1158.7 1154.0    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1158.7 1154.0    0.0 H/s
Highest:  1163.4 H/s
-----------------------------------------------------------------
```

Edited (Dual threaded) config 2.10.5:
```
  // gpu: gfx900  compute units: 56
  // memory:4048|4048|8048 MiB (used per thread|max per alloc|total free)
  { "index" : 1,
    "intensity" : 1792, "worksize" : 8,
    "affine_to_cpu" : false, "strided_index" : 0, "mem_chunk" : 2,
    "unroll" : 1, "comp_mode" : true, "interleave" : 40
  },
  { "index" : 1,
    "intensity" : 1792, "worksize" : 8,
    "affine_to_cpu" : false, "strided_index" : 0, "mem_chunk" : 2,
    "unroll" : 1, "comp_mode" : true, "interleave" : 40
  },
```
**Log:**
```
[2019-06-18 02:46:22] : OpenCL Interleave 1|0: 16/2409.32 ms - 40.5
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m | ID |    10s |    60s |    15m |
|  0 |  749.8 |   (na) |   (na) |  1 |  734.8 |   (na) |   (na) |
Totals (AMD):  1484.6    0.0    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1484.6    0.0    0.0 H/s
Highest:  1518.2 H/s
-----------------------------------------------------------------
[2019-06-18 02:47:23] : AMD GPU 1: Share accepted. Pool: pool.ryo-currency.ru:4444
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m | ID |    10s |    60s |    15m |
|  0 |  709.2 |  742.0 |   (na) |  1 |  830.0 |  756.4 |   (na) |
Totals (AMD):  1539.2 1498.3    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1539.2 1498.3    0.0 H/s
Highest:  1524.0 H/s
-----------------------------------------------------------------
[2019-06-18 02:47:27] : OpenCL Interleave 1|1: 43/2451.08 ms - 41.2
```

**Stak 2.8.0** (release binaries)
Auto generated config:
```
 // gpu: gfx900  compute units: 56
  // memory:4048|4048|8048 MiB (used per thread|max per alloc|total free)
  { "index" : 1,
    "intensity" : 1792, "worksize" : 8,
    "affine_to_cpu" : false, "strided_index" : 0, "mem_chunk" : 2,
    "unroll" : 8, "comp_mode" : true, "interleave" : 40
  },
```
**Log:**
```
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m |
|  0 | 1395.5 |   (na) |   (na) |
Totals (AMD):  1395.5    0.0    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1395.5    0.0    0.0 H/s
Highest:  1495.8 H/s
-----------------------------------------------------------------
[2019-06-18 02:38:53] : Result accepted by the pool.
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m |
|  0 | 1455.2 | 1448.2 |   (na) |
Totals (AMD):  1455.2 1448.2    0.0 H/s
-----------------------------------------------------------------
Totals (ALL):   1455.2 1448.2    0.0 H/s
Highest:  1495.8 H/s
```

_p.s changing unroll to 8 in latest binaries doen't help._
_p.s 2 versions 2.10.4 and lower are affected to low hr too._

**Comment by lordhugo7880** [comment]
Does it help if you change worksize to 16 and interleave to 28?

**Comment by Fredar** [comment]
Indeed strange it produces a single thread config for vegas and VII, same here.
Works fine with dual but requires manual change.

**Comment by Ryo-RU** [comment]
increasing autogenerated config `worksize` from 8 to 16 brings back lost hashrate

**Comment by Ryo-RU** [comment]
fixed in 2.10.7 release
